src/passes/rewrite-namespace-this-references.ts

```diff
@@ -19,13 +19,23 @@ import * as jsc from 'jscodeshift';
 
 /**
  * Rewrite `this` references to the explicit namespaceReference identifier
- * within a single BlockStatement. Don't traverse deeper into new scopes.
+ * within a single BlockStatement if they are the start of a member expression,
+ * otherwise they are rewritten to `undefined`. Don't traverse deeper into new
+ * scopes.
  */
 function rewriteSingleScopeThisReferences(
     blockStatement: estree.BlockStatement, namespaceReference: string) {
   astTypes.visit(blockStatement, {
     visitThisExpression(path: NodePath<estree.ThisExpression>) {
-      path.replace(jsc.identifier(namespaceReference));
+      const parent = path.parent;
+      if (parent && parent.node.type !== 'MemberExpression') {
```

Comment:
Note, this won't work for `this['foo-bar']`, but that's probably _really_ rare.

Reply:
I think it `MemberExpression` covers both `x.y` and `x['y']`: https://github.com/benjamn/ast-types/blob/2df3d43e3fdb736a74f4f2d40817f76beb4971f7/def/core.js#L306-L319

Reply:
Oh, but following from that, what happens to `ns['x']` when `ns` is actually removed?

Reply:
Good to know. And... yeah, that won't work.